src/shared/PropertyValueSetter.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    internal static class PropertyValueSetter
+    {
+        private static readonly MethodInfo CallPropertyAddRangeOpenGenericMethod =
+            typeof(PropertyValueSetter).GetMethod(nameof(CallPropertyAddRange));
+
+        public static void SetValue(
+            ModelMetadata metadata,
+            PropertyHelper propertyHelper,
```

Comment:
This makes it difficult to share this as public API. I could pass the setter and property info separately if we don't want to create shared src files.